src/stringnumber.js

```diff
@@ -38,25 +42,35 @@ export class StringNumber extends Modifier {
         pos = num.getPosition();
         const props = note.getKeyProps()[num.getIndex()];

```

Comment:
For me the real code smell occurs over the next three lines where the same `i` is reused in two internal looks -- it would seem to me that the outer loop is unnecessary (I'm sort of surprised that it works at all, but the inner loop leaves i == nums.length so it exits the outer loop when it leaves the inner loop.  I don't know if there's a linter in the world that would catch this, however.